examples/basecoin/app/app.go

```diff
@@ -32,19 +32,21 @@ type BasecoinApp struct {
 
 	// keys to access the substores
 	capKeyMainStore    *sdk.KVStoreKey
+	capKeyAccountStore *sdk.KVStoreKey
 	capKeyIBCStore     *sdk.KVStoreKey
 	capKeyStakingStore *sdk.KVStoreKey
 
 	// Manage getting and setting accounts
 	accountMapper sdk.AccountMapper
 }
 
-func NewBasecoinApp(logger log.Logger, db dbm.DB) *BasecoinApp {
+func NewBasecoinApp(logger log.Logger, dbMain, dbAcc, dbIBC, dbStaking dbm.DB) *BasecoinApp {
```

Comment:
lets start a convention here where we pass in a `map[string]dbm.DB` and then we can use that to fill in the capKeys, rather than this hardcoded list.

Then we can add more dbs (governance, etc.) without breaking the API

Reply:
Once #532 is merged we will only need one database and it feels strange that we want to pass in multiple databases.  I think it provides a strange API if we allow a map of databases to be passed in, because then we also need to have some convention around how those map to capKeys.

Reply:
The convention is extremely simple: use the same string. 

I have the following facts:

- the DB will be fixed an unknown amount of time in the future (call it months)
- we will add new stores soon (call it weeks)
- we have users of this API (including ourselves, in tests)
- the current approach will lead to more breakages of this API

I would think the above signals strongly that we should take an approach that can last for a bit and will require the least amount of maintenance work. Why set us up to do more unnecessary work in the future ?